modules/host/negotiaterevisecontract_test.go

```diff
@@ -0,0 +1,88 @@
+package host
+
+import (
+	"testing"
+)
+
+func TestShannonEntropy(t *testing.T) {
+	t.Parallel()
+	uniform := make([]byte, 256)
+	for i := 0; i < 256; i++ {
+		uniform[i] = byte(i)
+	}
+	almost := make([]byte, 256)
+	for i := 0; i < 256-10; i++ {
+		almost[i] = byte(i)
+	}
+	testCases := []struct {
+		name      string
+		input     []byte
+		low, high float64
+	}{
+		{
+			name:  "all zeros",
+			input: make([]byte, 100),
+			low:   0.0,
+			high:  0.00000001,
+		},
+		{
+			name:  "uniform distrinution",
+			input: uniform,
+			low:   0.9999999,
+			high:  1.0,
+		},
+		{
+			name:  "almost uniform distrinution",
+			input: almost,
+			low:   0.98,
+			high:  0.99,
+		},
```

Comment:
should also test random data (using `fastrand.Bytes()`

Reply:
Added, but `math/rand` was used instead of `fastrand` to make the input the same in all test invocations (and the test - reproducible).